src/tests/class_tests/openms/source/MRMTransitionGroupPicker_test.cpp

```diff
@@ -201,66 +270,99 @@ END_SECTION
+    sum = 0.0;
+    for (ConvexHull2D::PointArrayType::iterator it = data2_points.begin(); it != data2_points.end(); it++)
+    {
+      sum += it->getY();
+    }
+    // Part of the signal gets lost due to the re-sampling since the MS1 sampling
+    // positions are not at the same place as the MS2 sampling positions
+    double resampling_loss = 875.9514;
+    TEST_REAL_SIMILAR(sum, 53900 - resampling_loss);
+    TEST_REAL_SIMILAR(mrmfeature.getPrecursorFeature("Precursor_i0").getIntensity(),
+    // mrmfeature.getMS1Feature().getIntensity(), 53900 - resampling_loss);
+    53900 - resampling_loss);
   }
-  TEST_REAL_SIMILAR(sum, 59989.8287208466);
-  TEST_REAL_SIMILAR(mrmfeature.getFeature("1").getIntensity(), 59989.8287208466);
-
-  // Also check the MS1
-  std::vector<String> result;
-  mrmfeature.getPrecursorFeatureIDs(result);
-  TEST_EQUAL(result.size(), 1);
-  TEST_EQUAL(result[0], "Precursor_i0");
-  data2_points = mrmfeature.getPrecursorFeature("Precursor_i0").getConvexHulls()[0].getHullPoints();
-  sum = 0.0;
-  for (ConvexHull2D::PointArrayType::iterator it = data2_points.begin(); it != data2_points.end(); it++)
-  {
-    sum += it->getY();
+
+  { // transition group 2
+    MRMTransitionGroupType transition_group;
+    setup_transition_group2(transition_group);
+
+    MRMTransitionGroupPicker trgroup_picker;
+    Param picker_param = trgroup_picker.getDefaults();
+    picker_param.setValue("resample_boundary", 1.0);
+    picker_param.setValue("PeakPickerMRM:gauss_width", 10.0);
+    picker_param.setValue("PeakPickerMRM:peak_width", -1.0);
+    picker_param.setValue("PeakPickerMRM:signal_to_noise", 1.0);
+    picker_param.setValue("PeakPickerMRM:method", "corrected");
+    picker_param.setValue("use_consensus", "false");
+    trgroup_picker.setParameters(picker_param);
+    trgroup_picker.pickTransitionGroup(transition_group);
+
+    TEST_EQUAL(transition_group.getFeatures().size(), 2)
+    MRMFeature mrmfeature = transition_group.getFeatures()[0];
+    TEST_REAL_SIMILAR(mrmfeature.getRT(), 14);
+    TEST_REAL_SIMILAR(mrmfeature.getMetaValue("leftWidth"), 7);
+    TEST_REAL_SIMILAR(mrmfeature.getMetaValue("rightWidth"), 21);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("1").getIntensity(), 140);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("2").getIntensity(), 58); // consensus = 117
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("3").getIntensity(), 45);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("1").getMetaValue("peak_apex_int"), 16);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("2").getMetaValue("peak_apex_int"), 11);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("3").getMetaValue("peak_apex_int"), 3);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("1").getMetaValue("peak_apex_position"), 14);
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("2").getMetaValue("peak_apex_position"), 16); // consensus = 7
+    TEST_REAL_SIMILAR(mrmfeature.getFeature("3").getMetaValue("peak_apex_position"), 7);
   }
```

Comment:
can you test the same data with and without consensus?

Reply:
Test added in commit dd28aa1